modules/wildfly/src/main/java/org/testcontainers/containers/wildfly/WildFlyAdapter.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2019 IBM Corporation and others
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.testcontainers.containers.wildfly;
+
+import org.microshed.testing.testcontainers.spi.ServerAdapter;
+import org.testcontainers.images.builder.ImageFromDockerfile;
+
+import java.io.File;
+import java.util.Optional;
+
+public class WildFlyAdapter implements ServerAdapter {
+
+    @Override
+    public int getPriority() {
+        return PRIORITY_RUNTIME_MODULE;
+    }
+
+    @Override
+    public int getDefaultHttpPort() {
+        return 8080;
+    }
+
+    @Override
+    public int getDefaultHttpsPort() {
+        return 8443;
+    }
+
+    @Override
+    public ImageFromDockerfile getDefaultImage(File appFile) {
+        String appName = appFile.getName();
+        // Compose a docker image equivalent to doing:
+        // FROM jboss/wildfly:18.0.1.Final
+        // ADD target/myservice.war /opt/jboss/wildfly/standalone/deployments
+        ImageFromDockerfile image = new ImageFromDockerfile()
+                .withDockerfileFromBuilder(builder -> builder.from("jboss/wildfly:18.0.1.Final")
+                        .add(appName, "/opt/jboss/wildfly/standalone/deployments")
+                        .build())
+                .withFileFromFile(appName, appFile);
+        return image;
+
+    }
+
+    @Override
+    public Optional<String> getReadinessPath() {
+        return Optional.of("/health");
```

Comment:
Do you know if the /health endpoint is always enabled in wildfly 18.0.1.Final? Also, does it seem so be pretty consistent that the /health endpoint only returns HTTP 200 when the app is ready to serve requests?

Reply:
thanks for this hint. I checked it and in the [documentation](https://docs.wildfly.org/18/Admin_Guide.html#MicroProfile_Health_SmallRye) they say it's _included in the standalone configurations included in the WildFly distribution._ But while searching I found that the `/health` endpoint is served from the management port of WildFly (9990 by default). I think this won't work with the current setup. I could think of introducing a management port attribute within `ServerAdapter` or adjusting `getReadinessPath()` to return a `URL` ?

Reply:
It's interesting that the documentation says the module is enabled "standalone" deployments, but then it goes on to describe how to add the module manually, which implies that there can be images of Wildfly that do not have the health module enabled?
Just to be safe lets return an empty optional here for now, perhaps we can rework the ServerAdapter readiness path later on to incorporate a port

Reply:
Yes, I was also a little bit confused while reading the docs. I'll update the Adapter to the default behaviour in the interface